crates/router/src/core/payments.rs

```diff
@@ -1279,6 +1282,14 @@ where
                 &Some(merchant_connector_account.clone()),
             )?;
 
+            let apple_pay_flow =
+                decide_apple_pay_flow(payment_method_type, is_apple_pay_predecrypt);
```

Comment:
Return an `Option<ApplePayFlow>` from `is_applepay_predecrypt` function and use it in subsequent code. No need for another function.